lib/cisco_node_utils/syslog_server.rb

```diff
@@ -0,0 +1,109 @@
+# Syslog Server provider class
+#
+# Jonathan Tripathy et al., September 2015
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.join(File.dirname(__FILE__), 'node')
+require File.join(File.dirname(__FILE__), 'interface')
+
+module Cisco
+  # NtpServer - node utility class for NTP Server configuration management
+  class SyslogServer < NodeUtil
+    attr_reader :name, :level, :vrf
+
+    def initialize(syslogserver_name,
```

Comment:
What's the design rationale behind prepending 'syslogserver_' to the parameter names and instance variable names?

If you instead do:

```
def initialize(name, level, vrf, instantiate)
  @name = name
   @vrf = vrf
   @level = level
...
```

then the code is more readable, plus you don't have to custom-implement the name/level/vrf getter methods below.